### PR TITLE
Book details

### PR DESCRIPTION
**Pull request by coonik**
Book details

### CODE REVIEW

File: Assignment/Book-Management/src/app/core/services/book.service.ts [hunk 1]
```diff
@@ -4,20 +4,30 @@ import { Injectable, ElementRef } from '@angular/core';
...
-  constructor(private http: HttpClient, private route: Router) {}
+  constructor(private http: HttpClient, private route: Router, private authorService: AuthorService, private categoryService: CategoryService) {}
   getObs: Observable<any>;
   totalPages: number;
   totalItems: number;
   searchResult: [];
+  author: object = {};
+  categories: object[] = [];
+
+  token = "Bearer eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJuYW1laWQiOiIxIiwidW5pcXVlX25hbWUiOiJtYW5hZ2VyMSIsInJvbGUiOiJNQU5BR0VSIiwibmJmIjoxNTk1ODEyMjY2LCJleHAiOjE1OTYyNDQyNjYsImlhdCI6MTU5NTgxMjI2Nn0.7CiOI-95Vxr6PsrK3-Qo0CaZONrHefist8u8Jf4Wk-rdl68wpabj-qJN7OgdLXRbdIGlU8D3Rg8Cx4hYZuXDGw"
```
Comment (tranmanhlinh) [inline Assignment/Book-Management/src/app/core/services/book.service.ts:24]: Store token in localstorage.

File: Assignment/Book-Management/src/app/core/services/book.service.ts [hunk 2]
```diff
@@ -29,7 +39,89 @@ export class BookService {
...
+    let data = {
+      "name": book.bookName,
+      "description": book.description === "None" ? null : book.description,
+      "price": +book.price,
+      "year": +book.year,
+      "author": this.author,
+      "publisher": book.publisher,
+      "cover": null,
+      "categories": this.categories
+    }
```
Comment (tranmanhlinh) [inline Assignment/Book-Management/src/app/core/services/book.service.ts:111]: Dont use string quote for object keys if there is not necessary.

File: Assignment/Book-Management/src/app/core/services/category.service.ts [hunk 3]
```diff
@@ -4,11 +4,22 @@ import { environment } from './../../../environments/environment';
 
 @Injectable({providedIn: 'root'})
 export class CategoryService {
+
+  token = "Bearer eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJuYW1laWQiOiIxIiwidW5pcXVlX25hbWUiOiJtYW5hZ2VyMSIsInJvbGUiOiJNQU5BR0VSIiwibmJmIjoxNTk1ODEyMjY2LCJleHAiOjE1OTYyNDQyNjYsImlhdCI6MTU5NTgxMjI2Nn0.7CiOI-95Vxr6PsrK3-Qo0CaZONrHefist8u8Jf4Wk-rdl68wpabj-qJN7OgdLXRbdIGlU8D3Rg8Cx4hYZuXDGw"
+
```
Comment (tranmanhlinh) [inline Assignment/Book-Management/src/app/core/services/category.service.ts:9]: Store token in the localStorage and add to the request by Interceptors.

File: Assignment/Book-Management/src/app/modules/book/components/book-detail/book-detail.component.ts [hunk 4]
```diff
@@ -1,15 +1,224 @@
...
+    let categoriesId: number[] = [];
+    val.categories.forEach(val => {
+      categoriesId.push(val.id)
+    })
+
+    this.bookService.getAuthor(val.author.id);
+    this.bookService.getCategories(categoriesId);
+
+    console.log(val);
+
```
Comment (tranmanhlinh) [inline Assignment/Book-Management/src/app/modules/book/components/book-detail/book-detail.component.ts:87]: Remove console.log.

Comment (coonik) [reply Assignment/Book-Management/src/app/modules/book/components/book-detail/book-detail.component.ts]: done

File: Assignment/Book-Management/src/app/core/services/book.service.ts (hunk 1 above)
Comment (coonik) [reply Assignment/Book-Management/src/app/core/services/book.service.ts]: done

File: Assignment/Book-Management/src/app/core/services/category.service.ts (hunk 3 above)
Comment (coonik) [reply Assignment/Book-Management/src/app/core/services/category.service.ts]: done

**Pull request merged**